src/controller/CHIPDeviceController.cpp

```diff
@@ -602,13 +616,31 @@ ControllerDeviceInitParams DeviceController::GetControllerDeviceInitParams()
     };
 }
 
+#define TRAMPOLINE(Name, ArgType)                                                                                                  \
+    void DeviceCommissioner::On##Name(void * context, ArgType arg1)                                                                \
+    {                                                                                                                              \
+        ChipLogProgress(Controller, "Received %s callback", "Name");                                                               \
+        DeviceCommissioner * commissioner = static_cast<DeviceCommissioner *>(context);                                            \
+        commissioner->On##Name(arg1);                                                                                              \
+    }
+
+#define TRAMPOLINE_2(Name, ArgType, ArgType2)                                                                                      \
```

Comment:
TRAMPOLINE_2 and 4 are called only once - using macros makes the code harder to follow. How about inlining them?

I am actually wondering if to just copy & paste these methods instead of usin TRAMPOLINE here to have less macro usage. Macros make code search hard (I will not be able to search for On... stuff) and are harder to update. Also wondering here if "ChipLogProgress" actually prints out the name or just says `Name` as it is unclear from the macro.

Reply:
I have removed the macros.